test/test_spinning.py.in

```diff
@@ -90,41 +83,24 @@ def _raw_to_png(csv_data, png_path):
     plt.savefig(png_path[:-4] + "_resident_anonymous_memory.png")
 
 
-def on_process_start(process_started, launch_context, observation_process):
-    SetEnvironmentVariable(
-        'PROCESS_UNDER_TEST_ID', str(process_started.pid)).execute(launch_context)
-    observation_process.execute(launch_context)
-
 def generate_test_description(ready_fn):
     node_spinning_log_prefix = tempfile.mkstemp(
         prefix='overhead_node_test_results_@TEST_NAME@_', text=True)[1]
 
-    node_metrics_collector = ExecuteProcess(
-      cmd=[
-          os.path.join(
-              ament_index_python.get_package_prefix('buildfarm_perf_tests'),
-              'lib/buildfarm_perf_tests',
-              'system_metric_collector',
-          ),
-          # Arguments
-          "--timeout", "@NODE_SPINNING_TIMEOUT@",
-          "--log", node_spinning_log_prefix,
-          "--process_pid", EnvironmentVariable('PROCESS_UNDER_TEST_ID')
-      ],)
-
     node_spinning_test = Node(
         package='buildfarm_perf_tests',
         node_executable='node_spinning',
         output='log',
         sigterm_timeout=LaunchConfiguration('sigterm_timeout', default=@NODE_SPINNING_TIMEOUT@+2)
     )
 
+    node_metrics_collector, node_spinning_test_hook = attach_system_metric_collector(
+        node_spinning_test, node_spinning_log_prefix, timeout=@NODE_SPINNING_TIMEOUT@)
+
     return LaunchDescription([
-      RegisterEventHandler(OnProcessStart(
-          target_action=node_spinning_test,
-          on_start=lambda p, c: on_process_start(p, c, node_metrics_collector)
-        )),
+        node_spinning_test_hook,
         node_spinning_test,
+        launch_testing.util.KeepAliveProc(),
```

Comment:
This was also missing. Without it, there was actually a race - if `NodeSpinningTestResults` took more than about half of a second, the test would fail.